tests/lightning_qubit2/test_expval_2.py

```diff
@@ -0,0 +1,360 @@
+# limitations under the License.
+"""
+Tests for process and execute (expval calculation).
+"""
+# pylint: disable=too-many-arguments, redefined-outer-name
+import numpy as np
+import pennylane as qml
+import pytest
+from conftest import LightningDevice, THETA, PHI, VARPHI
+from pennylane.devices import DefaultQubit
+
+from pennylane_lightning.lightning_qubit import LightningQubit2
+
+if LightningDevice != LightningQubit2:
+    pytest.skip("Exclusive tests for lightning.qubit2. Skipping.", allow_module_level=True)
+
+if not LightningQubit2._CPP_BINARY_AVAILABLE:  # pylint: disable=protected-access
+    pytest.skip("No binary module found. Skipping.", allow_module_level=True)
+
+
+@pytest.fixture(params=[np.complex64, np.complex128])
+def dev(request):
+    return LightningQubit2(wires=3, c_dtype=request.param)
+
+
+def calculate_reference(tape):
+    dev = DefaultQubit(max_workers=1)
+    program, _ = dev.preprocess()
+    tapes, transf_fn = program([tape])
+    results = dev.execute(tapes)
+    return transf_fn(results)
+
+
+def process_and_execute(dev, tape):
+    program, _ = dev.preprocess()
+    tapes, transf_fn = program([tape])
+    results = dev.execute(tapes)
+    return transf_fn(results)
+
+
+@pytest.mark.parametrize("theta, phi", list(zip(THETA, PHI)))
+class TestExpval:
+    """Test expectation value calculations"""
+
+    def test_Identity(self, theta, phi, dev, tol):
+        """Tests applying identities."""
+
+        ops = [
+            qml.Identity(0),
+            qml.Identity((0, 1)),
+            qml.Identity((1, 2)),
+            qml.RX(theta, 0),
+            qml.RX(phi, 1),
+        ]
+        measurements = [qml.expval(qml.PauliZ(0))]
+        tape = qml.tape.QuantumScript(ops, measurements)
+
+        result = dev.execute(tape)
+        expected = np.cos(theta)
+        tol = 1e-5 if dev.c_dtype == np.complex64 else 1e-7
+
+        assert np.allclose(result, expected, tol)
+
+    def test_identity_expectation(self, theta, phi, dev, tol):
+        """Tests identity expectations."""
+
+        tape = qml.tape.QuantumScript(
+            [qml.RX(theta, wires=[0]), qml.RX(phi, wires=[1]), qml.CNOT(wires=[0, 1])],
+            [qml.expval(qml.Identity(wires=[0])), qml.expval(qml.Identity(wires=[1]))],
+        )
+        result = dev.execute(tape)
+
+        tol = 1e-5 if dev.c_dtype == np.complex64 else 1e-7
+
+        assert np.allclose(1.0, result, tol)
+
+    def test_multi_wire_identity_expectation(self, theta, phi, dev, tol):
+        """Tests multi-wire identity."""
+
+        tape = qml.tape.QuantumScript(
+            [qml.RX(theta, wires=[0]), qml.RX(phi, wires=[1]), qml.CNOT(wires=[0, 1])],
+            [qml.expval(qml.Identity(wires=[0, 1]))],
+        )
+        result = dev.execute(tape)
+        tol = 1e-5 if dev.c_dtype == np.complex64 else 1e-7
+
+        assert np.allclose(1.0, result, tol)
+
+    @pytest.mark.parametrize(
+        "wires",
+        [
+            ([0, 1]),
```

Comment:
Are negative numbers allowed as well? What about floats?

Reply:
They are. I will add test cases for them.